python/ray/data/_internal/logical/interfaces/logical_operator.py

```diff
@@ -51,3 +51,9 @@ def output_dependencies(self) -> List["LogicalOperator"]:
 
     def post_order_iter(self) -> Iterator["LogicalOperator"]:
         return super().post_order_iter()  # type: ignore
+
+    def schema(self):
```

Comment:
nit, better to make them an abstract method, to avoid forgetting to implement them in a sub class.